packages/web-app-files/src/composables/actions/files/useFileActionsCreateSpaceFromResource.ts

```diff
@@ -0,0 +1,123 @@
+import { Store } from 'vuex'
+import { computed, unref } from 'vue'
+import { useGettext } from 'vue3-gettext'
+import { FileAction, FileActionOptions } from 'web-pkg/src/composables/actions'
+import { useAbility, useClientService, useLoadingService, useRouter } from 'web-pkg/src/composables'
+import { isPersonalSpaceResource } from 'web-client/src/helpers'
+import { isLocationSpacesActive } from 'web-app-files/src/router'
+import { useCreateSpace } from 'web-app-files/src/composables/spaces'
+import { useSpaceHelpers } from 'web-pkg/src/composables/spaces'
+import PQueue from 'p-queue'
+
+export const useFileActionsCreateSpaceFromResource = ({ store }: { store?: Store<any> } = {}) => {
+  const { can } = useAbility()
+  const { $gettext, $ngettext } = useGettext()
+  const loadingService = useLoadingService()
+  const { createSpace } = useCreateSpace()
+  const { checkSpaceNameModalInput } = useSpaceHelpers()
+  const { webdav } = useClientService()
+  const router = useRouter()
+  const hasCreatePermission = computed(() => can('create-all', 'Space'))
+
+  const confirmAction = async ({ spaceName, resources, space }) => {
+    store.dispatch('hideModal')
+    const queue = new PQueue({ concurrency: 4 })
+    const copyOps = []
+
+    try {
+      const createdSpace = await createSpace(spaceName)
+      store.commit('runtime/spaces/UPSERT_SPACE', createdSpace)
+
+      if (resources.length === 1 && resources[0].isFolder) {
+        //If a single folder is selected we copy it's content to the Space's root folder
+        resources = (await webdav.listFiles(space, { path: resources[0].path })).children
+      }
+
+      for (const resource of resources) {
+        copyOps.push(
+          queue.add(() => webdav.copyFiles(space, resource, createdSpace, { path: resource.name }))
+        )
+      }
+
+      await Promise.all(copyOps)
+      store.dispatch('Files/resetFileSelection')
+      store.dispatch('showMessage', {
+        title: $gettext('Space was created successfully')
+      })
+    } catch (error) {
+      console.error(error)
+      store.dispatch('showMessage', {
+        title: $gettext('Creating space failed…'),
+        status: 'danger'
+      })
```

Comment:
Hm, what if the space was created partially? 
Like 3 out of 5 files were copied: do we want to keep the broken space or would it potentially make sense to delete it?
I would really like to avoid that users somehow miss the error message, see the space was created and then delete the source files because they think their data is in the space now.

@tbsbdr what do you think?

Reply:
I created https://github.com/owncloud/web/issues/8775 to follow up on this (very valid!) point. However, IMO that's not a merge blocker for now as we're heading towards the finish line for this sprint.